python/sdk/test/model_test.py

```diff
@@ -381,6 +381,188 @@ def test_create_prediction_job(self, version):
+                      body=json.dumps(job_1.to_dict()),
+                      status=200,
+                      content_type='application/json')
+
+        # Patch the method as currently it is not supported in the library
+        # https://github.com/getsentry/responses/issues/135
+        def _find_match(self, request):
+            for match in self._urls:
+                if request.method == match['method'] and \
+                        self._has_url_match(match, request.url):
+                    return match
+
+        def _find_match_patched(self, request):
+            for index, match in enumerate(self._urls):
+                if request.method == match['method'] and \
+                        self._has_url_match(match, request.url):
+                    if request.method == "GET" and request.url == "/v1/models/1/versions/1/jobs/1":
+                        return self._urls.pop(index)
+                    else:
+                        return match
+        responses._find_match = types.MethodType(_find_match_patched, responses)
+
+        for i in range(3):
+            responses.add("GET", '/v1/models/1/versions/1/jobs/1',
+                          body=json.dumps(job_1.to_dict()),
+                          status=500,
+                          content_type='application/json')
+
+        responses.add("GET", '/v1/models/1/versions/1/jobs/1',
+                      body=json.dumps(job_1.to_dict()),
+                      status=200,
+                      content_type='application/json')
+
+        job_1.status = "failed"
+        for i in range(5):
+            responses.add("GET", '/v1/models/1/versions/1/jobs/1',
+                          body=json.dumps(job_1.to_dict()),
+                          status=500,
+                          content_type='application/json')
+
+        bq_src = BigQuerySource(table="project.dataset.source_table",
+                                features=["feature_1", "feature2"],
+                                options={"key": "val"})
+
+        bq_sink = BigQuerySink(table="project.dataset.result_table",
+                               result_column="prediction",
+                               save_mode=SaveMode.OVERWRITE,
+                               staging_bucket="gs://test",
+                               options={"key": "val"})
+
+        job_config = PredictionJobConfig(source=bq_src,
+                                         sink=bq_sink,
+                                         service_account_name="my-service-account",
+                                         result_type=ResultType.INTEGER)
+
+        with pytest.raises(ValueError):
+            j = version.create_prediction_job(job_config=job_config)
+            assert j.id == job_1.id
+            assert j.error == job_1.error
+            assert j.name == job_1.name
+
+        # unpatch
+        responses._find_match = types.MethodType(_find_match, responses)
+        assert len(responses.calls) == 10
```

Comment:
Why are there 10 calls?

Reply:
1 call for the post-call creating the job
3 calls for the first 3 errors
1 success calls but pending
5 errors call 5XX

Is there any mismatch you found mas?